psfws/utils.py

```diff
@@ -0,0 +1,233 @@
+    """Return wind direction, in degrees, from u,v components of velocity."""
+    d = np.arctan2(y, x) * (180/np.pi)
+    return (d + 180) % 360
+
+
+def to_components(s, d):
+    """Calculate the wind velocity components given speed/direction."""
+    v = s * np.cos((d - 180) * np.pi/180)
+    u = s * np.sin((d - 180) * np.pi/180)
+    return {'v': v, 'u': u}
+
+
+def process_gfs(gfs_df):
+    """Return dataframe of processed global forecasting system data.
+
+    Input: dataframe of GFS obsevrations, columns = ['u', 'v']
+    Returns: dataframe of GFS observations, columns = ['u','v','speed','dir']
+
+    Processing steps:
+    - reverse u and v altitudes
+    - filter daytime datapoints
+    - add "speed" and "dir" columns
+    """
+    not_daytime = gfs_df.index.hour != 12
+    gfs_df = gfs_df.iloc[not_daytime].copy()
+    n = len(gfs_df)
+
+    # reverse, and disregard the top 5 altitudes
+    gfs_df['u'] = [gfs_df['u'].values[i][::-1][:-5] for i in range(n)]
+    gfs_df['v'] = [gfs_df['v'].values[i][::-1][:-5] for i in range(n)]
+
+    gfs_df['speed'] = [np.hypot(gfs_df['u'].values[i], gfs_df['v'].values[i])
+                       for i in range(n)]
+
+    gfs_df['dir'] = [to_direction(gfs_df['v'].values[i], gfs_df['u'].values[i])
+                     for i in range(n)]
+    return gfs_df
+
+
+def match_telemetry(telemetry, gfs_dates):
+    """Return speed, dir, and overlap datetimes between telemetry and GFS.
+
+    Telemetry vals returned are medians in 1h bins around each GFS output.
+    """
+    n_gfs = len(gfs_dates)
+
+    speed = telemetry['speed']
+    direction = telemetry['dir']
+
+    speed_ids = [speed.index[abs(gfs_dates[i] - speed.index)
+                             < pd.Timedelta('30min')] for i in range(n_gfs)]
+
+    dir_ids = [direction.index[abs(gfs_dates[i] - direction.index)
+                               < pd.Timedelta('30min')] for i in range(n_gfs)]
+
+    ids_to_keep = [i for i in range(n_gfs)
+                   if len(speed_ids[i]) != 0 and len(dir_ids[i]) != 0]
+
+    matched_s = [np.median(speed.loc[speed_ids[i]])
+                 for i in range(n_gfs) if i in ids_to_keep]
+    matched_d = [np.median(direction.loc[dir_ids[i]])
+                 for i in range(n_gfs) if i in ids_to_keep]
+
+    return np.array(matched_s), np.array(matched_d), gfs_dates[ids_to_keep]
+
+
+def interpolate(x, y, new_x, kind):
+    """Interpolate 1D array y at values x to new_x values.
+
+    Parameters
+    ----------
+    x : array
+        x values of input y
+    y : array
+        1D array to interpolate
+    new_x : array
+        x values of desired interpolated output
+    kind : str
+        Perform either 'cubic' or 'gp' interpolation
+
+    Returns
+    -------
+        new_y, interpolated values of y at positions new_x
+
+    """
+    if kind == 'cubic':
+        f_y = interp1d(x, y, kind='cubic')
+        new_y = f_y(new_x)
+    elif kind == 'gp':
+        from sklearn import gaussian_process
```

Comment:
sklearn is missing from the requirement list in the README

Note you could avoid making it a hard requirement while still being safe-ish. You could check whether the import works, and if not print a warning and switch to cubic.

I think it's probably easier (and cleaner) to just consider it a requirement though, especially since gp is the default for wind interpolation.